### ISSUE TITLE
Issue with setting Y1 and Y2 axis limit on Docx Chart

### ISSUE THREAD

**Issue by shaileshshettigar1998** [jsreport/jsreport-docx#36]
Basically, we are rendering multiple charts in loop and we are trying to set the Y1 and Y2 axis to start with value zero. But it looks like changes are being applied only to first chart in the loop and remaining charts taking axis limit values depending on the data.
Could you please help us with this.

Example : https://playground.jsreport.net/w/anon/XIG7q~Bl

**Comment by bjrmatos** [comment]
the reason why the rest of the charts in the loop does not start with value with zero is because internally the document does not have any min value set, even if you changing manually from Word properties of chart, Word decides when it makes sense to actually store the min value, it looks at the data of the chart and decides based on it. 

the best way to enforce a real min value for the axis is to set it with chart options, so you need to set the [scales object](https://jsreport.net/learn/docx#docxchart:~:text=supported%20options%20are%3A-,scales%20(object),-%2D%3E%20Options%20to%20control).

you need to pass an options object for each chart, so your data can look like this:

```json
{
    "monthly_avail_chart_list": [
  {
    "chart_data": {
      "datasets": [
        {
          "label": "Total Energy Production",
          "data": [
            "844.4",
            "860.1",
            "852.9",
            "994.1",
            "994.7",
            "1041.2",
            "1063.3",
            "1020.0",
            "1045.5",
            "1066.1",
            "1111.5",
            "1032.0",
            "1101.8",
            "1138.0",
            "1137.7"
          ]
        },
        {
          "label": "Average Availability",
          "data": [
            "74.7",
            "78.1",
            "79.0",
            "89.0",
            "89.2",
            "90.7",
            "93.3",
            "94.0",
            "94.5",
            "96.1",
            "96.5",
            "97.4",
            "97.7",
            "97.8",
            "98.4"
          ]
        }
      ],
      "labels": [
        "C02",
        "H05",
        "I04",
        "H06a",
        "E01",
        "F04",
        "B01",
        "E05",
        "G01b",
        "H01",
        "L03",
        "H02",
        "C04",
        "J01",
        "G08"
      ]
    },
    "chart_options": {
      "scales": {
          "yAxes": [{
                "display": true,
                "ticks": {
                    "min": 0
                }
          }, {
                "display": true,
                "ticks": {
                    "min": 0
                }
          }]
      } 
    },
    "title": "Total Energy Production Vs Average Availability",
    "monthly_avail_x_axis": "Device",
    "monthly_avail_y1_axis": "Total Energy Production (MWh)",
    "monthly_avail_y2_axis": "Average Availability (%)"
  },
  {
    "chart_data": {
      "datasets": [
        {
          "label": "Total Energy Production",
          "data": [
            "1099.7",
            "1157.3",
            "1093.3",
            "1160.2",
            "1125.0",
            "1040.6",
            "1128.2",
            "1108.3",
            "1135.7",
            "1131.4",
            "1070.9",
            "1119.2",
            "1163.9",
            "1168.3",
            "1147.8"
          ]
        },
        {
          "label": "Average Availability",
          "data": [
            "98.5",
            "98.7",
            "99.4",
            "99.5",
            "99.5",
            "99.5",
            "99.6",
            "99.7",
            "99.7",
            "99.8",
            "99.8",
            "99.9",
            "99.9",
            "99.9",
            "99.9"
          ]
        }
      ],
      "labels": [
        "G03a",
        "A01",
        "G02",
        "I02",
        "F03",
        "H03",
        "H08",
        "H07",
        "I01",
        "L01",
        "F05",
        "H04",
        "I03",
        "A02",
        "B04"
      ]
    },
    "chart_options": {
      "scales": {
          "yAxes": [{
                "display": true,
                "ticks": {
                    "min": 0
                }
          }, {
                "display": true,
                "ticks": {
                    "min": 0
                }
          }]
      } 
    },
    "title": "Total Energy Production Vs Average Availability",
    "monthly_avail_x_axis": "Device",
    "monthly_avail_y1_axis": "Total Energy Production (MWh)",
    "monthly_avail_y2_axis": "Average Availability (%)"
  },
  {
    "chart_data": {
      "datasets": [
        {
          "label": "Total Energy Production",
          "data": [
            "1114.7",
            "1095.0",
            "1160.8",
            "1113.8",
            "1088.1",
            "1123.8",
            "1174.8",
            "1149.2",
            "1161.5",
            "1146.6",
            "1111.3",
            "1097.2",
            "1127.5"
          ]
        },
        {
          "label": "Average Availability",
          "data": [
            "99.9",
            "99.9",
            "99.9",
            "99.9",
            "99.9",
            "99.9",
            "99.9",
            "99.9",
            "99.9",
            "100.0",
            "100.0",
            "100.0",
            "100.0"
          ]
        }
      ],
      "labels": [
        "A04",
        "C03",
        "G04",
        "G05",
        "B03",
        "E02",
        "F01",
        "G07",
        "J03",
        "D01",
        "E04a",
        "E06",
        "F06"
      ]
    },
    "chart_options": {
      "scales": {
          "yAxes": [{
                "display": true,
                "ticks": {
                    "min": 0
                }
          }, {
                "display": true,
                "ticks": {
                    "min": 0
                }
          }]
      } 
    },
    "title": "Total Energy Production Vs Average Availability",
    "monthly_avail_x_axis": "Device",
    "monthly_avail_y1_axis": "Total Energy Production (MWh)",
    "monthly_avail_y2_axis": "Average Availability (%)"
  }
]
}
```

the important part to check is the addition of the `chart_options` for each item, it contains configuration to ensure both y axis in charts starts at 0.

of course you should modify your template.docx to pass the options, like this:

![Image](https://github.com/user-attachments/assets/096308a0-530c-4b30-b6fa-a4fe6f6f1672)

the result is this:

![Image](https://github.com/user-attachments/assets/8e0682c3-4725-43d7-be88-1afb4d67fc7e)

all charts y axis starts at 0 now.

TL;DR: you need to set the axis explicitly from data
